Civi/Membership/OrderCompleteSubscriber.php

```diff
@@ -206,13 +207,9 @@ private static function getRelatedMemberships(int $contributionID): array {
     if (empty($membershipIDs)) {
       return [];
     }
-    // We could combine this with the MembershipPayment.get - we'd
-    // need to re-wrangle the params (here or in the calling function)
-    // as they would then me membership.contact_id, membership.is_test etc
-    return civicrm_api3('Membership', 'get', [
-      'id' => ['IN' => $membershipIDs],
-      'return' => ['id', 'contact_id', 'membership_type_id', 'is_test', 'status_id', 'end_date'],
-    ])['values'];
+    return (array) Membership::get(FALSE)->addWhere('id', 'IN', $membershipIDs)
+      ->addSelect('*', 'status_id:name')
+      ->execute();
```

Comment:
@eileenmcnaughton my point here is that the neither the api3 nor the api4 call specify checkPermissions so both will use their defaults. The default for api3 was false and the default for api4 is true.

Reply:
@colemanw yep - but since I am specifically setting the permission to FALSE there the default is irrelevant?

Reply:
Oh right, I missed that.